src/methods/join.js

```diff
@@ -449,14 +449,18 @@ export class JoinImplementation {
     } = e2;
     invariant(result2.effects === e2);
 
+    let emptyEffects = construct_empty_effects(realm);
+
     let result = this.joinOrForkResults(realm, joinCondition, result1, result2, e1, e2);
     if (result1 instanceof AbruptCompletion) {
       if (!(result2 instanceof AbruptCompletion)) {
         invariant(result instanceof PossiblyNormalCompletion);
+        e2.generator = emptyEffects.generator;
```

Comment:
Hm, mutating shared state? Why is that okay?

Reply:
Are we at the point where we can enforce that a generator is only ever directly held by one Effects now?

Reply:
It is not OK, but right now it is necessary. Long term, I'd like to fix this. That will not be easy and is going to take quite a bit of preparation.